File: faturamento-service/Faturamento.Api/Models/NotaFiscal.cs
namespace Faturamento.Api.Models;

public class NotaFiscal
{
    public int Id { get; set; }
    public int Numero { get; set; }
    public StatusNotaFiscal Status { get; set; } = StatusNotaFiscal.Aberta;
    public DateTime DataCriacao { get; set; } = DateTime.UtcNow;

    public List<ItemNotaFiscal> Itens { get; set; } = new();
}

public enum StatusNotaFiscal
{
    Aberta,
    Fechada
}

File: frontend/src/app/shared/models/NotaFiscal.ts
import { ItemNotaFiscal } from "./ItemNotaFiscal";

export interface NotaFiscal {
  id: number;
  numero: number;
  status: 'Aberta' | 'Fechada' | 'Processada' | 'Processando';
  dataCriacao: string;
  itens: ItemNotaFiscal[];
}